app/assets/javascripts/discourse/app/components/sidebar.js

```diff
@@ -35,5 +37,13 @@ export default class Sidebar extends GlimmerComponent {
     if (this.site.mobileView) {
       document.removeEventListener("click", this.collapseSidebar);
     }
+    this.customSections.forEach((customSection) => customSection.teardown());
```

Comment:
I'm curious if we need this when each customSection is a component and can specify their own `willDestroy` hook.

Reply:
Previously we had will Destroy, but I noticed it is never triggered. It it because customSection is POJO inherited from `BaseCustomSidebarSection`. That's why I decided to follow your teardown pattern

Reply:
O sorry, in this case we need to define a `willDestroy` hook in `<Sidebar::Section>` component and pass in the function that the `willDestroy` hook will call.